lib/db/memory.js

```diff
@@ -133,6 +150,47 @@ MemoryDB.prototype.query = function(collection, query, fields, options, callback
   });
 };
 
+MemoryDB.prototype.startTransaction = function(id, callback) {
+  if (this._transactions[id]) callback(new Error('Transaction already started'));
+  this._transactions[id] = this._transactions[id] || {};
+  this._transactions[id].commits = [];
+  callback();
+};
+
+MemoryDB.prototype.restartTransaction = function(id, callback) {
+  delete this._transactions[id];
+  this.startTransaction(id, callback);
+};
+
+MemoryDB.prototype.commitTransaction = function(id, callback) {
+  var transaction = this._transactions[id];
+  if (!transaction) return callback();
+
+  delete this._transactions[id];
+
+  // Heavy-handed but effective approach to synchronous transaction writing:
+  // let's just save the DB state before and restore if the transaction fails
+  var docs = JSON.stringify(this.docs);
+  var ops = JSON.stringify(this.ops);
+
+  for (var commit of transaction.commits) {
+    var error = commit();
+
+    if (error) {
+      this.docs = JSON.parse(docs);
+      this.ops = JSON.parse(ops);
```

Comment:
With potentially large sets of docs/ops in tests, using a structured clone would be better for performance/memory. You mentioned rfdc is working well for you in other situations, and I've looked at in the past and it seems good.